packages/data-stores/src/add-ons/hooks/use-available-storage-add-ons.ts

```diff
@@ -1,38 +1,50 @@
 import { useMemo } from '@wordpress/element';
-import { useSiteMediaStorage } from '../../site';
-import { STORAGE_LIMIT } from '../constants';
+import { SiteMediaStorage, useSiteMediaStorage } from '../../site';
+import { AddOnMeta } from '../types';
 import useStorageAddOns from './use-storage-add-ons';
 
 interface Props {
 	siteId?: number | null;
 }
 
+/**
+ * Check if the quantity for a storage add-on is available for purchase.
+ * @param quantity The number of gigabytes the given add-on adds to the site's storage
+ * @param storage Data returned from
+ */
+export function isStorageQuantityAvailable(
+	quantity: number,
+	storage?: SiteMediaStorage
+): boolean {
+	if ( ! storage ) {
+		return true;
+	}
+
+	const currentMaxStorage = storage.maxStorageBytes / Math.pow( 1024, 3 );
+	const maxStorageExcludingAddons = storage.maxStorageBytesExcludingAddons / Math.pow( 1024, 3 );
+	const existingAddOnStorage = currentMaxStorage - maxStorageExcludingAddons;
+
+	return existingAddOnStorage < quantity;
```

Comment:
I'm not sure how big of a problem this is in practice, but I take your point that we might as well return `existingAddOnStorage` directly from the API endpoint, since that's what we ultimately want anyway. I've updated the back-end PR (now at https://github.a8c.com/Automattic/wpcom/pull/168696).

Let me know your thoughts!